tests/determinism/test

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+. "$(dirname "$0")"/../common.sh
+
+# Test that verifies TASTy files can be generated deterministically
+echo "Testing TASTy file generation with Scala 3..."
```

Comment:
It's not customary to `echo` a description of the test like this, but, given that individual test output is silenced by `scripts/test.sh`, it's a welcome addition.